Sourcery/Parsing/Parser.swift

```diff
@@ -95,14 +95,37 @@ final class Parser {
     ///   - existingTypes: List of existing types to use for further parsing.
     /// - Returns: All types we could find.
     /// - Throws: parsing errors.
-    public func parseContents(_ contents: String, existingTypes: [Type] = []) -> [Type] {
+    public func parseContents(_ contents: String, existingTypes: [Type] = []) -> (types: [Type], typealiases: [String: String]) {
         guard !contents.hasPrefix(Sourcery.generationMarker) else {
             if verbose { print("\(logPrefix)Skipping source file because it was generated by Sourcery") }
-            return []
+            return ([], [:])
         }
 
         self.contents = contents
-        return parseTypes(Structure(file: File(contents: contents)).dictionary, existingTypes: existingTypes)
+        
+        let file = File(contents: contents)
+        let types = parseTypes(Structure(file: file).dictionary, existingTypes: existingTypes)
+        let typealiases = parseTypeAliases(SyntaxMap(file: file).tokens)
+        return (types, typealiases)
+    }
+    
+    internal func parseTypeAliases(_ tokens: [SyntaxToken]) -> [String: String] {
+        var typealiases = [String: String]()
+        tokens.enumerated()
+            .filter { _, token in
+                guard token.type == "source.lang.swift.syntaxtype.keyword" else { return false }
+                guard let keyword = contents.bridge().substringWithByteRange(start: token.offset, length: token.length) else { return false }
+                return keyword == "typealias"
+            }
+            .map({ (index, _) in (tokens[index + 1], tokens[index + 2]) })
+            .forEach({ (aliasToken, typeToken) in
+                guard
+                    let alias = contents.bridge().substringWithByteRange(start: aliasToken.offset, length: aliasToken.length),
+                    let type = contents.bridge().substringWithByteRange(start: typeToken.offset, length: typeToken.length) else { return }
+                
+                typealiases[alias] = type
```

Comment:
For now type aliases will not be accessible in templates, I simply use them here to find out the correct type information for variables and to extend proper class in case there is an extension on type alias.